insecure/corruptible/factory.go

```diff
@@ -178,6 +191,30 @@ func (c *ConduitFactory) processAttackerMessage(msg *insecure.Message) error {
 		return fmt.Errorf("could not decode message: %w", err)
 	}
 
+	switch e := event.(type) {
+	case *flow.ExecutionReceipt:
+		if len(e.ExecutorSignature) == 0 {
```

Comment:
is this something that will always be true, or could it be specific to the attack orchestrator? Wondering if this check belongs in the `generateExecutionReceipt` implementation

Reply:
It is specific to the attack orchestrator basically deciding whether to generate an execution receipt and override the current event with it, or just to pass the current event as it is without any modification. Hence, the logic is better to be separated from the creational pattern of `generateExecutionReceipt` that is responsible to generate an execution receipt out of the given result. 

It also applies a better level of information hiding and encapsulation, i.e., `generateExecutionReceipt` only receives the result part of the receipt, and does not need to be concern about other parts of that, while the inspection of the receipt is happening at the orchestrator level.